src/molecules/ToggleSwitch/ToggleSwitch.test.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ToggleSwitch from './ToggleSwitch';
+
+describe('Toggle Switch works properly', () => {
+  it('lets you toggle the switch', () => {
+    render(<ToggleSwitch />);
+    const toggle = screen.getByRole('checkbox', { id: /toggle/i });
+    userEvent.click(toggle);
```

Comment:
Same here, assert the state before the event, so you make sure this actually happens after you click the toggle.